src/context/index.ts

```diff
@@ -19,23 +20,67 @@ const nonPrimitiveProps = [
   'INCLUDED_PROPS'
 ];
 
-export default async function(config) {
+type Config = {
+  AUTH0_DOMAIN: string
+  AUTH0_CLIENT_ID: string
+  AUTH0_CLIENT_SECRET: string
+  AUTH0_INPUT_FILE: string
+  AUTH0_ACCESS_TOKEN?: string
+  AUTH0_AUDIENCE?: string
+  AUTH0_API_MAX_RETRIES?: number
+}// TODO
+
+type Asset = { [key: string]: any }
+
+export type Assets = {
+  actions: Asset[],
+  attackProtection: Asset,
+  clients: Asset[],
+  clientGrants: Asset[],
+  connections: Asset[],
+  databases: Asset[],
+  emailProvider: Asset,
+  emailTemplates: Asset[],
+  guardianFactorProviders: Asset[],
+  guardianFactors: Asset[],
+  guardianFactorTemplates: Asset[],
+  guardianPhoneFactorMessageTypes: Asset[],
+  guardianPhoneFactorSelectedProvider: Asset,
+  guardianPolicies: Asset[],
+  hooks: Asset[],
+  migrations: Asset[]
+  organizations: Asset[],
+  pages: Asset[],
+  resourceServers: Asset[],
+  roles: Asset[],
+  rules: Asset[],
+  rulesConfigs: Asset[],
+  tenant: Asset,
+  triggers: Asset[],
+  //non-resource types
+  exclude: {
+    [key: string]: string[]
+  },
+  clientsOrig: Asset[],
+}
```

Comment:
This `Assets` type is a bit unwieldy but it is fundamental to many of the core functions of this repo. Like the `Config` type, there is a task to clear this a bit more in the future.